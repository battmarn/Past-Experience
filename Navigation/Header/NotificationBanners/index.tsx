import React, { type FC, useRef } from 'react';
import { useSelector } from 'react-redux';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

import { NotificationBanner } from 'app/components/NotificationBanner';
import type { Notification } from 'app/reducers/notifications';

import s from './styles.module.css';

interface NotificationBannersProps {
  isNavbarMinimised?: boolean;
}

const NotificationBannerItem: FC<{
  in?: boolean;
  notification: Notification;
  isNavbarMinimised: boolean | undefined;
}> = ({ in: inProp, notification, isNavbarMinimised }) => {
  const nodeRef = useRef<HTMLElement>(null);

  return (
    <CSSTransition
      in={inProp}
      appear
      key={notification.text}
      nodeRef={nodeRef}
      unmountOnExit
      timeout={{
        enter: 500,
        exit: 500,
      }}
      classNames={{
        enter: s.enter,
        enterActive: s.enterActive,
        enterDone: s.enterDone,
        exit: s.exit,
        exitActive: s.exitActive,
        exitDone: s.exitDone,
      }}
    >
      <NotificationBanner ref={nodeRef} isNavbarMinimised={isNavbarMinimised} inCheckout={false} {...notification} />
    </CSSTransition>
  );
};

export const NotificationBanners: FC<NotificationBannersProps> = ({ isNavbarMinimised }) => {
  const notifications = useSelector((state) => state.notifications) || [];

  return (
    <TransitionGroup className={s.banners}>
      {notifications.map((notification) => {
        return (
          <NotificationBannerItem
            key={notification.text}
            notification={notification}
            isNavbarMinimised={isNavbarMinimised}
          />
        );
      })}
    </TransitionGroup>
  );
};
