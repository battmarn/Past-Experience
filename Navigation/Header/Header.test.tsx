import * as useMediaQuery from '@react-hook/media-query';
import { trackStructEvent } from '@snowplow/browser-tracker';
import type { RegionString } from '@trinnylondon/global.types.config';
import React from 'react';
import * as reactRedux from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { screen, userEvent, within } from 'app/bit-migration/test-utilities/react-renderer';
import { createUseSignOutMock } from 'app/hooks/auth/signOut.mock';
import { createUseCurrentLocationMock } from 'app/hooks/useCurrentLocationMock';
import { useScrollDirection } from 'app/hooks/useScrollDirection';
import { createUseSiteIdMock } from 'app/hooks/useSiteIdMock';
import { clearDataLayer, snapshotDataLayer } from 'app/lib/analytics/analytics.mock';
import { createUseFlagValueMock } from 'app/lib/featureFlag.mock';
import { createUseTrackingLocationMock } from 'app/providers/GTM/TrackingLocationProvider.mock';
import { AlgoliaSearchProvider } from 'app/providers/search/AlgoliaSearchProvider';
import { ffSearchConfig } from 'config/featureFlag';
import { renderWithStore } from 'test/helpers/renderWithStore';

import { NavigationContextProvider } from '../Context/Context';
import { mockCategory, mockCategorySkincare } from '../mocks';

import { Header, type HeaderProps } from './Header';

jest.mock('app/hooks/useScrollDirection');
jest.mock('@snowplow/browser-tracker', () => ({
  trackStructEvent: jest.fn(),
}));

const setup = (props: HeaderProps) => {
  return renderWithStore(
    <NavigationContextProvider>
      <Header {...props} />
    </NavigationContextProvider>,
  );
};

describe('<Header />', () => {
  const testProps: HeaderProps = {
    categories: [mockCategory, mockCategorySkincare],
    mobileMenuOpen: false,
    closeDesktopNav: jest.fn(),
    setActiveCategory: jest.fn(),
    setCategoryMenu: jest.fn(),
    isCheckout: false,
    isNavbarMinimised: false,
    isProcessingPayment: false,
  };

  beforeEach(() => {
    clearDataLayer();
    createUseSignOutMock();
    createUseSiteIdMock('uk' as RegionString);
    createUseTrackingLocationMock('Header');

    jest.spyOn(useMediaQuery, 'useMediaQuery').mockImplementation(() => true);
    (useScrollDirection as jest.Mock).mockReturnValue('start');
  });

  describe('for non-checkout pages', () => {
    it('should contain a labelled button to open the search menu', () => {
      createUseFlagValueMock({
        [ffSearchConfig]: {
          debounceMs: 0,
          maxQueryLength: 9,
          minQueryLength: 2,
        },
      });
      createUseCurrentLocationMock({
        pathname: 'uk',
      });

      const { container } = renderWithStore(
        <BrowserRouter>
          <NavigationContextProvider>
            <AlgoliaSearchProvider apiKey="123456">
              <Header {...testProps} />
            </AlgoliaSearchProvider>
          </NavigationContextProvider>
        </BrowserRouter>,
      );
      const searchTrigger = within(container).getByRole('button', { name: 'Open search menu', expanded: false });
      expect(searchTrigger).toBeInTheDocument();
    });
    it('will render the header component for non checkout pages', () => {
      createUseCurrentLocationMock({
        pathname: 'uk',
      });

      expect(setup(testProps).container).toMatchSnapshot();
    });

    describe('mobile menu', () => {
      it('will render the mobile menu open icon when mobile menu is closed', () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });

        setup(testProps);
        expect(screen.getByTestId('mobile-menu-open')).toBeInTheDocument();
      });

      it('will render the mobile menu closed icon when the mobile menu is open', () => {
        const openTestProps = {
          ...testProps,
          mobileMenuOpen: true,
        };
        createUseCurrentLocationMock({
          pathname: 'uk',
        });

        setup(openTestProps);
        expect(screen.getByTestId('mobile-menu-close')).toBeInTheDocument();
      });

      it('will toggle the mobile menu open when the mobile menu button is clicked', async () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });
        const dispatchMock = jest.fn();
        jest.spyOn(reactRedux, 'useDispatch').mockReturnValue(dispatchMock);

        setup(testProps);
        await userEvent.click(screen.getByTestId('mobile-nav-toggle'));
        expect(dispatchMock).toHaveBeenCalledWith({
          type: 'OPEN_NAVIGATION',
        });
        snapshotDataLayer(1);
      });

      it('will toggle the mobile menu closed when the mobile menu button is clicked', async () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });
        const dispatchMock = jest.fn();
        jest.spyOn(reactRedux, 'useDispatch').mockReturnValue(dispatchMock);

        const openProps = {
          ...testProps,
          mobileMenuOpen: true,
        };
        setup(openProps);
        await userEvent.click(screen.getByTestId('mobile-nav-toggle'));
        expect(dispatchMock).toHaveBeenCalledWith({
          type: 'CLOSE_NAVIGATION',
        });
        snapshotDataLayer(1);
      });
    });

    describe('site banners', () => {
      const siteBannerProps: HeaderProps = {
        ...testProps,
        siteBanners: <div />,
      };
      it('will update css classes when site banners are available', () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });

        expect(setup(siteBannerProps).container).toMatchSnapshot();
      });
    });

    describe('search trigger', () => {
      it('will render a button to open the search menu', () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });
        setup(testProps);
        expect(screen.getByRole('button', { name: /open/i })).toBeInTheDocument();
      });
    });

    describe('logo', () => {
      it('will render the non checkout logo', () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });
        setup(testProps);
        expect(screen.getByTestId('non-checkout-logo')).toBeInTheDocument();
      });

      it('will track the logo click', async () => {
        createUseCurrentLocationMock({
          pathname: 'uk',
        });
        setup(testProps);
        await userEvent.click(screen.getByTestId('non-checkout-logo'));
        snapshotDataLayer(1);

        expect(trackStructEvent).toHaveBeenCalledWith({
          category: 'Site interaction',
          action: 'Clicked logo',
          label: 'Header',
        });
      });
    });

    it('will render the basket component', () => {
      createUseCurrentLocationMock({
        pathname: 'uk',
      });
      setup(testProps);
      expect(screen.getByTestId('basket-desktop')).toBeInTheDocument();
    });
  });

  describe('for checkout pages', () => {
    const checkoutProps = {
      ...testProps,
      queryString: '/uk/checkout/',
    };

    it('will render the header component for checkout', () => {
      createUseCurrentLocationMock({
        pathname: 'uk/checkout/',
      });

      expect(setup(checkoutProps).container).toMatchSnapshot();
    });

    describe('site banners', () => {
      it('will render the site banners container when site banners are present', () => {
        const props: HeaderProps = {
          ...checkoutProps,
          siteBanners: <div data-test-id="site-banners" />,
        };
        createUseCurrentLocationMock({
          pathname: '/uk/checkout/',
        });
        setup(props);
        expect(screen.getByTestId('site-banners-container')).toBeInTheDocument();
        expect(screen.getByTestId('site-banners')).toBeInTheDocument();
      });
    });
  });
});
