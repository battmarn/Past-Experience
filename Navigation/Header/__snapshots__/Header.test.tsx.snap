// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> for checkout pages will render the header component for checkout 1`] = `
<div>
  <header
    class="header"
    data-header="header"
    data-test-id="header-not-minimised"
  >
    <div
      class="site-banners"
      data-test-id="site-banners-container"
    >
      <div
        class="banner-currency-container"
      >
        <div
          class="border banner-currency"
          role="menu"
          tabindex="0"
        >
          <div
            aria-hidden="true"
            class="desktopSelect hidden"
            data-test-id="country-select"
            role="menu"
            tabindex="0"
          >
            <p
              class="menuShippingText"
            >
              site-menu-header
            </p>
            <a
              class="menuOption"
              data-test-id="desktop-currency-menu-option"
              href="de/checkout/"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/de.svg"
              />
              EUR | €
            </a>
            <a
              class="menuOption"
              data-test-id="desktop-currency-menu-option"
              href="se/checkout/"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/se.svg"
              />
              SEK | kr
            </a>
            <a
              class="menuOption active"
              data-test-id="desktop-currency-menu-option"
              href="uk/checkout/"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/uk.svg"
              />
              GBP | £
            </a>
          </div>
          <div
            class="flagsContainer"
          >
            <button
              class="buttonSelect"
              data-test-id="desktop-currency-button-select"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/uk.svg"
              />
              <span
                data-test-id="header-selected-site"
              >
                GBP | £
              </span>
              <img
                alt="arrow"
                class="arrows"
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="top-line"
    >
      <div
        class="headerIconButtons first"
      >
        <div
          class="mobile-menu-button"
        >
          <button
            class="accessibleFocus"
            data-test-id="mobile-nav-toggle"
            type="button"
          >
            <test-file-stub
              classname="icon btn-icon"
              data-test-id="mobile-menu-open"
            />
          </button>
        </div>
        <div
          role="search"
        >
          <button
            aria-expanded="false"
            aria-label="Open search menu"
            class="searchTriggerButton accessibleFocus"
            type="button"
          >
            <test-file-stub
              aria-label="Open search menu"
              classname="icon btn-icon"
            />
            <div
              class="searchHint"
            >
              <div
                class="searchHint"
              >
                <span
                  class="searchHintText"
                >
                  search
                </span>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="headerLogo"
      >
        <div
          class="logoContainer"
        >
          <a
            class="logoLink accessibleFocus MuiBox-root emotion-0"
            data-test-id="non-checkout-logo"
            href="/"
          >
            <h1
              class="homePageLogoH1"
            >
              <picture>
                <source
                  media="(max-width: 519px)"
                  srcset="test-file-stub"
                />
                <img
                  alt="Trinny London"
                  class="logo"
                  src="test-file-stub"
                />
              </picture>
              <span
                class="visuallyHidden"
              >
                Trinny London
              </span>
            </h1>
          </a>
        </div>
      </div>
      <div
        class="headerIconButtons last"
      >
        <div
          class="account"
        >
          <button
            aria-expanded="false"
            class="accessibleFocus"
            data-test-id="navigation-account"
          >
            <test-file-stub
              aria-label="My Account"
              classname="icon accountIcon"
            />
          </button>
          <div
            class="accountHoverDropdown isInvisible"
            data-test-id="navigation-account-dropdown"
            hidden=""
            role="menu"
          >
            <span
              class="accountName"
              data-test-id="navigation-account-dropdown-name"
            >
              hello, undefined
            </span>
            <ul
              class="accountLinksList"
              data-test-id="account-dropdown-links"
            >
              <li
                class="accountLinkItem"
              >
                <a
                  class="accountLinkButton MuiBox-root emotion-0"
                  data-test-id="account-link-profile"
                  href="/my-account"
                >
                  <span>
                    My account
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="accountLinkItem"
              >
                <a
                  class="accountLinkButton MuiBox-root emotion-0"
                  data-test-id="account-link-help"
                  href="https://trinny-london.zendesk.com/hc/en-uk"
                >
                  <span>
                    Help
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="accountLinkItem"
              >
                <button
                  class="accountLinkButton"
                  data-test-id="account-link-signout"
                >
                  <span>
                    Sign out
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="basket-button"
          data-test-id="basket-desktop"
        >
          <button
            aria-label="View your bag"
            data-test-id="navigation-cart"
          >
            <div
              class="iconWrapper filled"
            >
              <test-file-stub
                aria-label="Cart"
                classname="icon cartIcon"
              />
              <span
                class="text navText itemCount"
                data-test-id="bag-item-count"
              >
                2
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="minimised-site-banners"
  />
  <div
    class="fader-container"
  >
    <nav
      aria-labelledby=":rb:"
      class="nav-menu"
      data-test-id="navbar"
    >
      <h2
        class="visuallyHidden"
        id=":rb:"
      >
        Main Menu
      </h2>
      <div
        class="navbar-list first"
      >
        <div
          class="navbar-list-item"
          data-test-id="category-item-1ShH2Iklr6mypdfqwf4EOQ"
        >
          <button
            class="animateUnderline"
            data-test-id="category-item-button-1ShH2Iklr6mypdfqwf4EOQ"
          >
            Makeup
          </button>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="logo-middle-container"
      >
        <a
          class="logo-link MuiBox-root emotion-0"
          data-test-id="minimised-logo"
          href="/"
        >
          <img
            alt=""
            aria-hidden="true"
            class="logo-middle-img"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="navbar-list second"
      >
        <div
          class="navbar-list-item"
          data-test-id="category-item-3A29nMpiiTsTshdybf8OJ0"
        >
          <button
            class="animateUnderline"
            data-test-id="category-item-button-3A29nMpiiTsTshdybf8OJ0"
          >
            Skincare
          </button>
        </div>
      </div>
    </nav>
    <div
      class="fader"
    />
  </div>
  <div
    class="banners"
  />
</div>
`;

exports[`<Header /> for non-checkout pages logo will track the logo click 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "image click",
      "location": "Header",
      "name": "Clicked - Logo",
    },
  },
]
`;

exports[`<Header /> for non-checkout pages mobile menu will toggle the mobile menu closed when the mobile menu button is clicked 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Closed Mobile Nav",
    },
  },
]
`;

exports[`<Header /> for non-checkout pages mobile menu will toggle the mobile menu open when the mobile menu button is clicked 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Mobile Nav",
    },
  },
]
`;

exports[`<Header /> for non-checkout pages site banners will update css classes when site banners are available 1`] = `
<div>
  <header
    class="header"
    data-header="header"
    data-test-id="header-not-minimised"
  >
    <div
      class="site-banners"
      data-test-id="site-banners-container"
    >
      <div />
      <div
        class="banner-currency-container"
      >
        <div
          class="border banner-currency"
          role="menu"
          tabindex="0"
        >
          <div
            aria-hidden="true"
            class="desktopSelect hidden"
            data-test-id="country-select"
            role="menu"
            tabindex="0"
          >
            <p
              class="menuShippingText"
            >
              site-menu-header
            </p>
            <a
              class="menuOption"
              data-test-id="desktop-currency-menu-option"
              href="de"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/de.svg"
              />
              EUR | €
            </a>
            <a
              class="menuOption"
              data-test-id="desktop-currency-menu-option"
              href="se"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/se.svg"
              />
              SEK | kr
            </a>
            <a
              class="menuOption active"
              data-test-id="desktop-currency-menu-option"
              href="uk"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/uk.svg"
              />
              GBP | £
            </a>
          </div>
          <div
            class="flagsContainer"
          >
            <button
              class="buttonSelect"
              data-test-id="desktop-currency-button-select"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/uk.svg"
              />
              <span
                data-test-id="header-selected-site"
              >
                GBP | £
              </span>
              <img
                alt="arrow"
                class="arrows"
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="top-line top-line-margin"
    >
      <div
        class="headerIconButtons first"
      >
        <div
          class="mobile-menu-button"
        >
          <button
            class="accessibleFocus"
            data-test-id="mobile-nav-toggle"
            type="button"
          >
            <test-file-stub
              classname="icon btn-icon"
              data-test-id="mobile-menu-open"
            />
          </button>
        </div>
        <div
          role="search"
        >
          <button
            aria-expanded="false"
            aria-label="Open search menu"
            class="searchTriggerButton accessibleFocus"
            type="button"
          >
            <test-file-stub
              aria-label="Open search menu"
              classname="icon btn-icon"
            />
            <div
              class="searchHint"
            >
              <div
                class="searchHint"
              >
                <span
                  class="searchHintText"
                >
                  search
                </span>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="headerLogo"
      >
        <div
          class="logoContainer"
        >
          <a
            class="logoLink accessibleFocus MuiBox-root emotion-0"
            data-test-id="non-checkout-logo"
            href="/"
          >
            <picture>
              <source
                media="(max-width: 519px)"
                srcset="test-file-stub"
              />
              <img
                alt="Trinny London"
                class="logo"
                src="test-file-stub"
              />
            </picture>
          </a>
        </div>
      </div>
      <div
        class="headerIconButtons last"
      >
        <div
          class="account"
        >
          <button
            aria-expanded="false"
            class="accessibleFocus"
            data-test-id="navigation-account"
          >
            <test-file-stub
              aria-label="My Account"
              classname="icon accountIcon"
            />
          </button>
          <div
            class="accountHoverDropdown isInvisible"
            data-test-id="navigation-account-dropdown"
            hidden=""
            role="menu"
          >
            <span
              class="accountName"
              data-test-id="navigation-account-dropdown-name"
            >
              hello, undefined
            </span>
            <ul
              class="accountLinksList"
              data-test-id="account-dropdown-links"
            >
              <li
                class="accountLinkItem"
              >
                <a
                  class="accountLinkButton MuiBox-root emotion-0"
                  data-test-id="account-link-profile"
                  href="/my-account"
                >
                  <span>
                    My account
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="accountLinkItem"
              >
                <a
                  class="accountLinkButton MuiBox-root emotion-0"
                  data-test-id="account-link-help"
                  href="https://trinny-london.zendesk.com/hc/en-uk"
                >
                  <span>
                    Help
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="accountLinkItem"
              >
                <button
                  class="accountLinkButton"
                  data-test-id="account-link-signout"
                >
                  <span>
                    Sign out
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="basket-button"
          data-test-id="basket-desktop"
        >
          <button
            aria-label="View your bag"
            data-test-id="navigation-cart"
          >
            <div
              class="iconWrapper filled"
            >
              <test-file-stub
                aria-label="Cart"
                classname="icon cartIcon"
              />
              <span
                class="text navText itemCount"
                data-test-id="bag-item-count"
              >
                2
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="minimised-site-banners"
  />
  <div
    class="fader-container"
  >
    <nav
      aria-labelledby=":r6:"
      class="nav-menu"
      data-test-id="navbar"
    >
      <h2
        class="visuallyHidden"
        id=":r6:"
      >
        Main Menu
      </h2>
      <div
        class="navbar-list first"
      >
        <div
          class="navbar-list-item"
          data-test-id="category-item-1ShH2Iklr6mypdfqwf4EOQ"
        >
          <button
            class="animateUnderline"
            data-test-id="category-item-button-1ShH2Iklr6mypdfqwf4EOQ"
          >
            Makeup
          </button>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="logo-middle-container"
      >
        <a
          class="logo-link MuiBox-root emotion-0"
          data-test-id="minimised-logo"
          href="/"
        >
          <img
            alt=""
            aria-hidden="true"
            class="logo-middle-img"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="navbar-list second"
      >
        <div
          class="navbar-list-item"
          data-test-id="category-item-3A29nMpiiTsTshdybf8OJ0"
        >
          <button
            class="animateUnderline"
            data-test-id="category-item-button-3A29nMpiiTsTshdybf8OJ0"
          >
            Skincare
          </button>
        </div>
      </div>
    </nav>
    <div
      class="fader"
    />
  </div>
  <div
    class="banners"
  />
</div>
`;

exports[`<Header /> for non-checkout pages will render the header component for non checkout pages 1`] = `
<div>
  <header
    class="header"
    data-header="header"
    data-test-id="header-not-minimised"
  >
    <div
      class="site-banners"
      data-test-id="site-banners-container"
    >
      <div
        class="banner-currency-container"
      >
        <div
          class="border banner-currency"
          role="menu"
          tabindex="0"
        >
          <div
            aria-hidden="true"
            class="desktopSelect hidden"
            data-test-id="country-select"
            role="menu"
            tabindex="0"
          >
            <p
              class="menuShippingText"
            >
              site-menu-header
            </p>
            <a
              class="menuOption"
              data-test-id="desktop-currency-menu-option"
              href="de"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/de.svg"
              />
              EUR | €
            </a>
            <a
              class="menuOption"
              data-test-id="desktop-currency-menu-option"
              href="se"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/se.svg"
              />
              SEK | kr
            </a>
            <a
              class="menuOption active"
              data-test-id="desktop-currency-menu-option"
              href="uk"
              target="_top"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/uk.svg"
              />
              GBP | £
            </a>
          </div>
          <div
            class="flagsContainer"
          >
            <button
              class="buttonSelect"
              data-test-id="desktop-currency-button-select"
            >
              <img
                alt="uk flag"
                class="image"
                src="/assets/flags/uk.svg"
              />
              <span
                data-test-id="header-selected-site"
              >
                GBP | £
              </span>
              <img
                alt="arrow"
                class="arrows"
                src="test-file-stub"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="top-line"
    >
      <div
        class="headerIconButtons first"
      >
        <div
          class="mobile-menu-button"
        >
          <button
            class="accessibleFocus"
            data-test-id="mobile-nav-toggle"
            type="button"
          >
            <test-file-stub
              classname="icon btn-icon"
              data-test-id="mobile-menu-open"
            />
          </button>
        </div>
        <div
          role="search"
        >
          <button
            aria-expanded="false"
            aria-label="Open search menu"
            class="searchTriggerButton accessibleFocus"
            type="button"
          >
            <test-file-stub
              aria-label="Open search menu"
              classname="icon btn-icon"
            />
            <div
              class="searchHint"
            >
              <div
                class="searchHint"
              >
                <span
                  class="searchHintText"
                >
                  search
                </span>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="headerLogo"
      >
        <div
          class="logoContainer"
        >
          <a
            class="logoLink accessibleFocus MuiBox-root emotion-0"
            data-test-id="non-checkout-logo"
            href="/"
          >
            <picture>
              <source
                media="(max-width: 519px)"
                srcset="test-file-stub"
              />
              <img
                alt="Trinny London"
                class="logo"
                src="test-file-stub"
              />
            </picture>
          </a>
        </div>
      </div>
      <div
        class="headerIconButtons last"
      >
        <div
          class="account"
        >
          <button
            aria-expanded="false"
            class="accessibleFocus"
            data-test-id="navigation-account"
          >
            <test-file-stub
              aria-label="My Account"
              classname="icon accountIcon"
            />
          </button>
          <div
            class="accountHoverDropdown isInvisible"
            data-test-id="navigation-account-dropdown"
            hidden=""
            role="menu"
          >
            <span
              class="accountName"
              data-test-id="navigation-account-dropdown-name"
            >
              hello, undefined
            </span>
            <ul
              class="accountLinksList"
              data-test-id="account-dropdown-links"
            >
              <li
                class="accountLinkItem"
              >
                <a
                  class="accountLinkButton MuiBox-root emotion-0"
                  data-test-id="account-link-profile"
                  href="/my-account"
                >
                  <span>
                    My account
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="accountLinkItem"
              >
                <a
                  class="accountLinkButton MuiBox-root emotion-0"
                  data-test-id="account-link-help"
                  href="https://trinny-london.zendesk.com/hc/en-uk"
                >
                  <span>
                    Help
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="accountLinkItem"
              >
                <button
                  class="accountLinkButton"
                  data-test-id="account-link-signout"
                >
                  <span>
                    Sign out
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
                    data-test-id="icon-chevron-right"
                    focusable="false"
                    style="width: 11px; height: 11px;"
                    viewBox="0 0 5 10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 1L5 5L1 9"
                      stroke="currentColor"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="basket-button"
          data-test-id="basket-desktop"
        >
          <button
            aria-label="View your bag"
            data-test-id="navigation-cart"
          >
            <div
              class="iconWrapper filled"
            >
              <test-file-stub
                aria-label="Cart"
                classname="icon cartIcon"
              />
              <span
                class="text navText itemCount"
                data-test-id="bag-item-count"
              >
                2
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="minimised-site-banners"
  />
  <div
    class="fader-container"
  >
    <nav
      aria-labelledby=":r1:"
      class="nav-menu"
      data-test-id="navbar"
    >
      <h2
        class="visuallyHidden"
        id=":r1:"
      >
        Main Menu
      </h2>
      <div
        class="navbar-list first"
      >
        <div
          class="navbar-list-item"
          data-test-id="category-item-1ShH2Iklr6mypdfqwf4EOQ"
        >
          <button
            class="animateUnderline"
            data-test-id="category-item-button-1ShH2Iklr6mypdfqwf4EOQ"
          >
            Makeup
          </button>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="logo-middle-container"
      >
        <a
          class="logo-link MuiBox-root emotion-0"
          data-test-id="minimised-logo"
          href="/"
        >
          <img
            alt=""
            aria-hidden="true"
            class="logo-middle-img"
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="navbar-list second"
      >
        <div
          class="navbar-list-item"
          data-test-id="category-item-3A29nMpiiTsTshdybf8OJ0"
        >
          <button
            class="animateUnderline"
            data-test-id="category-item-button-3A29nMpiiTsTshdybf8OJ0"
          >
            Skincare
          </button>
        </div>
      </div>
    </nav>
    <div
      class="fader"
    />
  </div>
  <div
    class="banners"
  />
</div>
`;
